client/src/app/core/repositories/base-repository.ts

```diff
@@ -346,13 +354,32 @@ export abstract class BaseRepository<V extends BaseViewModel, M extends BaseMode
     /**
      * update the observable of the list. Also updates the sorting of the view model list.
      */
-    public commitUpdate(modelIds: Id[]): void {
+    public commitUpdate(modelIds: Id[], isFullList: boolean = false): void {
+        if (isFullList) {
+            this.doFullListUpdate(modelIds);
+        }
         this.unsafeViewModelListSubject.next(this.getViewModelListUnsafe());
         modelIds.forEach(id => {
             this.updateViewModelObservable(id);
         });
     }
 
+    protected doFullListUpdate(modelIds: Id[]): void {
+        const deletedIds = this.getViewModelIdList().difference(modelIds);
+        this.deleteModels(deletedIds);
+        deletedIds.forEach(id => this.updateViewModelObservable(id));
+        this.updateViewModelListSubject(Object.values(this.viewModelStore));
```

Comment:
Please undo this all. Small note: This would not be correct since the wull update is triggered in `commitUpdate` at the end - this would result in the observables triggering two times and the unsafe model list is not updated. This is the reason why it should be handled in the AU service: You do not have to change something here.